source/server/listener_impl.cc

```diff
@@ -589,7 +586,7 @@ void ListenerImpl::buildTlsInspectorListenerFilter() {
 
     auto& factory =
         Config::Utility::getAndCheckFactoryByName<Configuration::NamedListenerFilterConfigFactory>(
-            Extensions::ListenerFilters::ListenerFilterNames::get().TlsInspector);
+            "envoy.filters.listener.tls_inspector");
```

Comment:
I think this is the only name that is repeated, maybe extract this into a constant within this file? That way we avoid the compile dependency but we make sure the names within this file are consistent at least.